main.tf

```diff
@@ -11,47 +8,48 @@ locals {
   dummy = "dummy"
 }
 data "azurerm_key_vault" "payment_key_vault" {
-  name = "ccpay-${var.env}"
-  resource_group_name = "ccpay-${var.env}"
+  name = join("-", [var.core_product, var.env])
+  resource_group_name = join("-", [var.core_product, var.env])
 }
 
 data "azurerm_key_vault_secret" "s2s_client_secret" {
   name = "gateway-s2s-client-secret"
-  key_vault_id = "${data.azurerm_key_vault.payment_key_vault.id}"
+  key_vault_id = data.azurerm_key_vault.payment_key_vault.id
 }
 
 data "azurerm_key_vault_secret" "s2s_client_id" {
   name = "gateway-s2s-client-id"
-  key_vault_id = "${data.azurerm_key_vault.payment_key_vault.id}"
+  key_vault_id = data.azurerm_key_vault.payment_key_vault.id
 }
 
 data "template_file" "policy_template" {
   template = "${file("${path.module}/template/api-policy.xml")}"
 
-  vars {
-    allowed_certificate_thumbprints = "${local.thumbprints_in_quotes_str}"
-    s2s_client_id = "${data.azurerm_key_vault_secret.s2s_client_id.value}"
-    s2s_client_secret = "${data.azurerm_key_vault_secret.s2s_client_secret.value}"
-    s2s_base_url = "${local.s2sUrl}"
+  vars ={
+    allowed_certificate_thumbprints = local.thumbprints_in_quotes_str
+    s2s_client_id = data.azurerm_key_vault_secret.s2s_client_id.value
+    s2s_client_secret = data.azurerm_key_vault_secret.s2s_client_secret.value
+    s2s_base_url = local.s2sUrl
   }
 }
 
 data "template_file" "api_template" {
   template = "${file("${path.module}/template/api.json")}"
 }
 resource "azurerm_template_deployment" "api" {
```

Comment:
this should be using one of our terraform modules for api management not an arm template

Reply:
If this causes the resource to be deleted, we cannot make this change.